### PR TITLE
Allow creating `Vec`s of exported enums

### PR DESCRIPTION
**Pull request by cschramm**
Also implements `TryFrom<JsValue>` and `Into<JsValue>` for them.

Enums were left behind in #3554. This adds the missing bits.

### CODE REVIEW

**Review by daxpedda — APPROVED** [review]
LGTM!
As the author of the original implementation I'm gonna wait for @Liamolucko's approval.

File: crates/backend/src/codegen.rs
Comment (cschramm) [reply crates/backend/src/codegen.rs]: Oh, true. Fixed by replicating the cast clauses instead.

**Review by Liamolucko — APPROVED** [review]
Thanks!

**Comment by SIGSTACKFAULT** [comment]
when will this ship? wasm-bindgen hasn't had a release in months. This and #3554 would be really nice to be able to actually use

**Comment by daxpedda** [comment]
See https://github.com/rustwasm/wasm-bindgen/issues/3530.

**Pull request merged**
